SQLite/Core/Connection.swift

```diff
@@ -25,8 +25,209 @@
 import Dispatch
 import CSQLite
 
+
+/// The mode in which a transaction acquires a lock.
+public enum TransactionMode : String {
+    
+    /// Defers locking the database till the first read/write executes.
+    case Deferred = "DEFERRED"
+    
+    /// Immediately acquires a reserved lock on the database.
+    case Immediate = "IMMEDIATE"
+    
+    /// Immediately acquires an exclusive lock on all databases.
+    case Exclusive = "EXCLUSIVE"
+    
+}
+
+
+/// Protocol to an SQLite connection
+public protocol Connection {
```

Comment:
Given this I'd love to keep it as is.  `DBConnection` & `BorrowedConnection` (although you never use it directly) are more descriptive and seem to be more in line with the eventual Swift naming schemes; the alternative of `ConnectionProtocol`, `Connection` & `BorrowedConnection` seems less desirable to me even if it means a bit of renaming in projects.

In reality, projects shouldn't have to do much renaming at all since `Connection` just becomes a protocol.  Only during initialization & when using special functions would you need to use `DBConnection`.